packages/next/client/components/bailout-to-client-rendering.ts

```diff
@@ -0,0 +1,21 @@
+import { staticGenerationAsyncStorage } from './static-generation-async-storage'
+
+export const BAILOUT_TO_CLIENT_RENDERING_ERROR_CODE =
+  'BAILOUT_TO_CLIENT_RENDERING_ERROR_CODE'
+
+export function bailoutToClientRendering(reason: string) {
```

Comment:
```suggestion
export function bailoutToClientRendering(reason: string): true | never {
```

Reply:
Would be `true | undefined | never`. `undefined` if we're not statically generating. Worth adding? If so also here https://github.com/vercel/next.js/blob/canary/packages/next/client/components/static-generation-bailout.ts#L4.

Reply:
Maybe better to just return `boolean | never` instead?

Reply:
`boolean | never` added 🙏